src/module/data/actor/base.mjs

```diff
@@ -72,6 +72,22 @@ export default class BaseActorModel extends foundry.abstract.TypeDataModel {
     return 1;
   }
 
+  /**
+   * The actor's level
+   */
+  get level() {
+    return 1;
+  }
+  
+  /**
+   * The actor's echelon based on their current level
+   */
+  get echelon() {
+    return Object.entries(ds.CONFIG.echelons).reduce((acc, current) => {
```

Comment:
You can destructure `current` to something like `entry[key, value]` so you can just have `value.threshold` and `Number(key)`

Reply:
Thanks. Didn't think to destructure there